datafusion/functions/src/encoding/inner.rs

```diff
@@ -266,7 +266,7 @@ macro_rules! decode_to_array {
 
 impl Encoding {
     fn encode_scalar(self, value: Option<&[u8]>) -> ColumnarValue {
-        ColumnarValue::Scalar(match self {
+        ColumnarValue::from(match self {
```

Comment:
The switch from explicit use of a variant to a `From` clause is a nice change regardless, FWIW, and we could potentially break it out into its own PR